cloud/store/recover.go

```diff
@@ -0,0 +1,183 @@
+	Address  string `msgpack:"Address" json:"address"`
+	ID       string `msgpack:"ID" json:"id"`
+	Suffrage int64  `msgpack:"Suffrage" json:"-"`
+	NonVoter bool   `msgpack:"-" json:"non_voter"`
+}
+
+// genPeersFileFromBolt checks if there was an existing raft.db and has persisted config
+// so it generates peers.jon file to be used for raft.RecoverCluster to recover the cluster on start.
+func (st *Store) genPeersFileFromBolt(raftDBPath, peerFilePath string) error {
+	if _, err := os.Stat(raftDBPath); err == nil {
+		bdb, err := raftbolt.NewBoltStore(raftDBPath)
+		if err != nil {
+			return err
+		}
+		defer bdb.Close()
+
+		f, err := bdb.FirstIndex()
+		if err != nil {
+			return err
+		}
+		l, err := bdb.LastIndex()
+		if err != nil {
+			return err
+		}
+
+		st.log.Info(fmt.Sprintf("found previous config at %s, recovery will be initiated", raftDBPath))
+
+		log := new(raft.Log)
+		var peers []Peer
+		peersMap := make(map[string]Peer, l)
+		for i := f; i <= l; i++ {
+			err = bdb.GetLog(i, log)
+			if err != nil {
+				return err
+			}
+
+			// Unmarshal the MessagePack data into a map
+			var decodedData map[string]interface{}
+			err := msgpack.Unmarshal(log.Data, &decodedData)
+			if err != nil {
+				st.log.Error("unable to unmarshal msgpack from raft.db file err=" + err.Error())
+				continue
+			}
+
+			// Extract the "Servers" array from the map
+			servers, ok := decodedData["Servers"].([]interface{})
+			if !ok {
+				st.log.Error("failed to extract 'Servers' array")
+				continue
+			}
+
+			// Iterate over the array and decode each Peer
+			for _, server := range servers {
+				serverMap, ok := server.(map[string]interface{})
+				if !ok {
+					st.log.Error("Failed to convert server to map")
+					continue
+				}
+
+				var peer Peer
+				if address, ok := serverMap["Address"].(string); ok {
+					peer.Address = address
+				}
+				if id, ok := serverMap["ID"].(string); ok {
+					peer.ID = id
+				}
+				if suffrage, ok := serverMap["Suffrage"].(int64); ok {
+					peer.Suffrage = suffrage
+					peer.NonVoter = suffrage != int64(raft.Voter)
+				}
+
+				existedP := peersMap[peer.Address]
+				if reflect.DeepEqual(existedP, peer) {
+					continue
+				}
+				peersMap[peer.Address] = peer
+				peers = append(peers, peer)
+			}
+		}
+
+		if len(peers) == 0 {
+			return nil
+		}
+
+		configData, err := json.MarshalIndent(peers, "", "  ")
+		if err != nil {
+			return err
+		}
+
+		file, err := os.OpenFile(peerFilePath, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0o644)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		if _, err := file.Write(configData); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// configViaPeers checks if the peers.json file exists with the provided path
+// if the files exists it creates Raft.Configuration from it
+func (st *Store) configViaPeers(peerFilePath string) (raft.Configuration, error) {
+	var configuration raft.Configuration
+
+	if _, err := os.Stat(peerFilePath); err == nil {
```

Comment:
Same here we can return early and avoid a big indent block

Reply:
agreed